Gradle 6.3 incompatibility - Deprecation logging method moved

### Describe the bug

An internal deprecation logging method from gradle is referenced here:

https://github.com/Triple-T/gradle-play-publisher/blob/88485d158b940bed40fa22f8dd9a6ae92b118d53/play/plugin/src/main/kotlin/com/github/triplet/gradle/play/internal/Validation.kt#L30

The mechanism for emitting deprecation messages changed between gradle 6.2.2 and 6.3 https://github.com/gradle/gradle/pull/12265

### How To Reproduce

Use a PKCS12 authentication style (I know, I should change, but it's deprecated not removed yet...)

#### Versions

- Gradle Play Publisher: 2.7.3
- Gradle Wrapper: gradle version 6.3 (just released, <6.3 works)
- Android Gradle Plugin: 3.6.1

#### Tasks executed

Fails to load the build.gradle file, just doing `./gradlew wrapper` to update to gradle 6.3 triggered this, *assuming* you have a pkcs12 GCP config file vs JSON

### Expected behavior

Should work, with a deprecation message

### Additional context (if a crash, provide stack trace)

```
2020-03-31T13:17:20.263-0500 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Caused by: java.lang.NoSuchMethodError: org.gradle.internal.deprecation.DeprecationMessageBuilder.undocumented()Lorg/gradle/internal/deprecation/DeprecationMessageBuilder$WithDocumentation;
2020-03-31T13:17:20.263-0500 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at com.github.triplet.gradle.play.internal.ValidationKt.validateCreds(Validation.kt:30)
2020-03-31T13:17:20.263-0500 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at com.github.triplet.gradle.play.PlayPublisherPlugin$applyInternal$1.execute(PlayPublisherPlugin.kt:141)
2020-03-31T13:17:20.263-0500 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at com.github.triplet.gradle.play.PlayPublisherPlugin$applyInternal$1.execute(PlayPublisherPlugin.kt:52)
2020-03-31T13:17:20.263-0500 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.ImmutableActionSet$SingletonSet.execute(ImmutableActionSet.java:225)
2020-03-31T13:17:20.263-0500 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.DefaultDomainObjectCollection.doAdd(DefaultDomainObjectCollection.java:264)
2020-03-31T13:17:20.263-0500 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.DefaultDomainObjectCollection.add(DefaultDomainObjectCollection.java:253)
2020-03-31T13:17:20.263-0500 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at com.android.build.gradle.AppExtension.addVariant(AppExtension.java:84)
2020-03-31T13:17:20.263-0500 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at com.android.build.gradle.internal.ApiObjectFactory.create(ApiObjectFactory.java:127)
2020-03-31T13:17:20.263-0500 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   ... 143 more

```

I rely on this plugin and like it a lot, thanks for producing it. It's my hope this bug report is helpful and I'll have a potential patch in a moment

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.2.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.3-all.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists

--- a/play/plugin/src/main/kotlin/com/github/triplet/gradle/play/internal/Validation.kt
+++ b/play/plugin/src/main/kotlin/com/github/triplet/gradle/play/internal/Validation.kt
@@ -5,7 +5,6 @@ import com.github.triplet.gradle.common.validation.validateDebuggability
 import com.github.triplet.gradle.play.PlayPublisherExtension
 import com.github.triplet.gradle.play.PlayPublisherPlugin
 import org.gradle.api.logging.Logging
-import org.gradle.internal.deprecation.DeprecationLogger
 
 internal fun PlayPublisherExtension.validateCreds() {
     val creds = checkNotNull(config.serviceAccountCredentials) {
@@ -24,11 +23,11 @@ internal fun PlayPublisherExtension.validateCreds() {
         check(config.serviceAccountEmail != null) {
             "PKCS12 credentials must specify a service account email."
         }
-        DeprecationLogger.deprecate("Gradle Play Publisher's PKCS12 based authentication")
-                .withAdvice("Use JSON based authentication instead.")
-                .withContext("https://github.com/Triple-T/gradle-play-publisher#service-account")
-                .undocumented()
-                .nagUser()
+        Logging.getLogger(PlayPublisherPlugin::class.java)
+                .warn("Gradle Play Publisher's PKCS12 based authentication is deprecated.\n" +
+                      "This is scheduled to be removed in GPP 3.0.\n" +
+                      "Please use JSON based authentication instead.\n" +
+                      "https://github.com/Triple-T/gradle-play-publisher#service-account.")
     }
 }
 

--- a/play/plugin/src/test/kotlin/com/github/triplet/gradle/play/PlayPublisherPluginIntegrationTest.kt
+++ b/play/plugin/src/test/kotlin/com/github/triplet/gradle/play/PlayPublisherPluginIntegrationTest.kt
@@ -150,7 +150,7 @@ class PlayPublisherPluginIntegrationTest : IntegrationTestBase() {
             withArguments("help", "--warning-mode", "all")
         }
 
-        assertThat(result.output).contains("PKCS12 based authentication has been deprecated")
+        assertThat(result.output).contains("PKCS12 based authentication is deprecated")
     }
 
     @Test

